doc/contributing/erm-guidelines.md

```diff
@@ -0,0 +1,453 @@
+  }
+}
+```
+
+### A Note on documenting disposable objects
+
+When documenting disposable objects, it is important to clearly indicate that
+the object is disposable and how it should be disposed of. This includes
+documenting the `Symbol.dispose` and `Symbol.asyncDispose` methods, as well as
+any explicit disposal methods that the object exposes.
+
+If the disposable object is anonymous (that is, it is a regular JavaScript
+object that implements the `Symbol.dispose` method), it is still important to
+document that it is disposable and how it should be disposed of.
+
+Within the documentation, it is possible to documen anonymous objects as if
+they were classes, using the `Class: ` prefix and otherwise presenting the
+object as if you were documenting a regular JavaScript class, even if it is
+never actually instantiated as a class. Examples of this pattern can be seen,
+for instance, in the documentation of the Web Crypto API in
+`/doc/api/webcrypto.md`.
+
+So, for example, if you have an API that returns an anonymous disposable object,
+you might docucment it like:
+
+```markdown
+### Class: `MyDisposableObject`
+
+#### `myDisposableObject.dispose()`
+
+...
+
+#### `myDisposableObject[Symbol.dispose]()`
+
+...
+
+### `foo.getMyDisposableObject()`
+
+* Returns: {MyDisposableObject}
+```
+
+## Guidelines for Introducing explicit resource management into Existing APIs
+
+Introducing the ability to use `using` into existing APIs can be tricky.
+
+The best way to understand the issues is to look at a real world example. PR
+[58516](https://github.com/nodejs/node/pull/58516) is a good case. This PR
+sought to introduce `Symbol.dispose` and `Symbol.asyncDispose` capabilities
+into the `fs.mkdtemp` API such that a temporary directory could be created and
+be automatically disposed of when the scope in which it was created exited.
+However, the existing implementation of the `fs.mkdtemp` API returns a string
+value that cannot be made disposable. There are also sync, callback, and
+promise-based variations of the existing API that further complicate the
+situation.
+
+In the initial proposal, the `fs.mkdtemp` API was changed to return an object
+that implements the `Symbol.dispose` method but only if a specific option is
+provided. This would mean that the return value of the API would become
+polymorphic, returning different types based on how it was called. This adds
+a lot of complexity to the API and makes it difficult to reason about the
+return value. It also makes it difficult to programmatically detect whether
+the version of the API being used supports `using` or not.
+`fs.mkdtemp('...', { disposable: true })` would act differently in older versions
+of Node.js than in newer versions with no way to detect this at runtime other
+than to inspect the return value.
+
+Some APIs that already return objects that can be made disposable do not have
+this kind of issue. For example, the `setTimeout()` API in Node.js returns an
```

Comment:
This example might get stale soon - https://github.com/nodejs/node/issues/58689

Maybe [subprocess](https://nodejs.org/api/child_process.html#subprocesssymboldispose) or [readline](https://nodejs.org/api/readline.html#rlsymboldispose) instead?